src/containers/Cluster/Cluster.tsx

```diff
@@ -108,17 +111,20 @@ export function Cluster({
     }, [dispatch]);
 
     const actualClusterTabs = React.useMemo(() => {
-        let tabs = clusterTabs;
+        const skippedTabs: ClusterTab[] = [];
 
         if (!shouldShowNetworkTable) {
-            tabs = tabs.filter((tab) => tab.id !== clusterTabsIds.network);
+            skippedTabs.push(clusterTabsIds.network);
         }
         if (!shouldShowEventsTab) {
-            tabs = tabs.filter((tab) => tab.id !== clusterTabsIds.events);
+            skippedTabs.push(clusterTabsIds.events);
+        }
+        if (!isViewerUser) {
```

Comment:
Why don't you also check capabilities here?

Reply:
I'll clarify this on Monday!